### PR TITLE
Adjust the scheb/2fa-bundle integration

### PR DESCRIPTION
**Pull request by bytehead**
| Q                | A
| -----------------| ---
| Fixed issues     | Fixes #2680 
| Docs PR or issue | -

This PR addresses the comments made by @scheb in #2680 and is its follow up PR.

### CODE REVIEW

File: core-bundle/src/DependencyInjection/Security/ContaoLoginFactory.php [hunk 1]
```diff
@@ -53,19 +52,15 @@ protected function createAuthProvider(ContainerBuilder $container, $id, $config,
     {
         $twoFactorProviderId = TwoFactorFactory::PROVIDER_ID_PREFIX.$id;
         $twoFactorFirewallConfigId = 'contao.security.two_factor_firewall_config.'.$id;
-        $twoFactorFirewallConfig = new Definition(TwoFactorFirewallConfig::class, [$config, $id, null]);
 
         $container
-            ->setDefinition($twoFactorFirewallConfigId, $twoFactorFirewallConfig)
-            ->replaceArgument(2, new Reference('security.http_utils'))
+            ->setDefinition($twoFactorFirewallConfigId, new ChildDefinition(TwoFactorFirewallConfig::class))
```
Comment (scheb) [inline core-bundle/src/DependencyInjection/Security/ContaoLoginFactory.php]: Shouldn't that be `new ChildDefinition(TwoFactorFactory::FIREWALL_CONFIG_DEFINITION_ID)`?

Comment (bytehead) [reply core-bundle/src/DependencyInjection/Security/ContaoLoginFactory.php]: Yes, obviously 🙈 Thanks! Fixed in 094039a3b.

File: core-bundle/src/DependencyInjection/Security/ContaoLoginFactory.php [hunk 2]
```diff
@@ -53,19 +51,15 @@ protected function createAuthProvider(ContainerBuilder $container, $id, $config,
     {
         $twoFactorProviderId = TwoFactorFactory::PROVIDER_ID_PREFIX.$id;
         $twoFactorFirewallConfigId = 'contao.security.two_factor_firewall_config.'.$id;
-        $twoFactorFirewallConfig = new Definition(TwoFactorFirewallConfig::class, [$config, $id, null]);
```
Comment (aschempp) [inline core-bundle/src/DependencyInjection/Security/ContaoLoginFactory.php:56]: this removes `$config` and `$id` from the firewall config definition

**Review by aschempp — CHANGES_REQUESTED** [review]

File: core-bundle/src/DependencyInjection/Security/ContaoLoginFactory.php (hunk 2 above)
Comment (bytehead) [reply core-bundle/src/DependencyInjection/Security/ContaoLoginFactory.php:56]: True, fixed in 91d5177f7.

**Comment by leofeyer** [comment]
Thank you @bytehead.

**Pull request merged**